clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/VeniceVersionedStatsReporter.java

```diff
@@ -31,20 +27,9 @@ public VeniceVersionedStatsReporter(
 
     this.isSystemStore = VeniceSystemStoreUtils.isSystemStore(storeName);
 
-    /**
-     * All stats which are aware of version should line up with Helix listener. It's duplicate
-     * to store version metrics for each type of stats. It also causes metric conflict. See
-     * {@link MetricsRepository#registerMetric}. Adding an atomic state may not be a good
-     * way to deal with it. We might want to refactor the code in the future to remove it. (e.g.
-     * we could have a centralized stats version dispatcher that listens to Helix data changes,
-     * keeps track of store version and notifies each stats.
-     * TODO: get ride of {@link IS_VERSION_STATS_SETUP}
-     */
-    if (IS_VERSION_STATS_SETUP.compareAndSet(false, true)) {
-      registerSensor("current_version", new VersionStat(() -> (double) currentVersion));
-      if (!isSystemStore) {
-        registerSensor("future_version", new VersionStat(() -> (double) futureVersion));
-      }
+    if (!isSystemStore) {
```

Comment:
Shall we emit this for system store as well since it is very cheap?

Reply:
Sure